code/modules/clothing/under/miscellaneous.dm

```diff
@@ -691,12 +691,18 @@
 		if(extinguishes_left)
 			if(next_extinguish > world.time)
 				return
-			next_extinguish = world.time + extinguish_cooldown
-			extinguishes_left--
-			H.visible_message("<span class='warning'>[H]'s suit automatically extinguishes [H.p_them()]!</span>","<span class='warning'>Your suit automatically extinguishes you.</span>")
-			H.ExtinguishMob()
-			new /obj/effect/particle_effect/water(get_turf(H))
-	return 0
+				next_extinguish = world.time + extinguish_cooldown
+				extinguishes_left--
+			if((!istype(H.w_uniform, /obj/item/clothing/under/plasmaman/clown)))

```

Comment:
This basically comes down to an `istype(src, ...)` which should really be done by overriding `Extinguish` (or a new proc specifically for this check) for the clown suit instead

Reply:
Okay so how exactly would one override Extinguish? It just seems to result in a duplicate definition.

Reply:
This is `/obj/item/clothing/under/plasmaman/proc/Extinguish(mob/living/carbon/human/H)`.
When overriding, don't include the `proc` keyword. You want the clown suit to have special behavior, so put it under the clown clothing type:
`/obj/item/clothing/under/plasmaman/clown/Extinguish(mob/living/carbon/human/H)`.

Reply:
Ebic compiling and doing some thorough testing before commiting.